Dockerfile

```diff
@@ -1,39 +1,52 @@
 # Dockerfile for TA-Lib. To build:
 #
-#    docker build --rm -t talib -f Dockerfile .
+#    docker build --rm -t talib .
 #
 # To run:
 #
 #    docker run --rm -it talib bash
 #
-FROM python:3.6
 
+ARG PYTHON_VERSION="3.7"
+
+FROM python:$PYTHON_VERSION as builder
+
+ENV TA_PREFIX="/opt/ta-lib-core"
+ENV TA_LIBRARY_PATH="$TA_PREFIX/lib" \
+    TA_INCLUDE_PATH="$TA_PREFIX/include"
+
+WORKDIR /src/ta-lib-core
 RUN apt-get update && apt-get install -y \
-    gfortran \
-    libfreetype6-dev \
-    libhdf5-dev \
-    liblapack-dev \
-    libopenblas-dev \
-    libpng-dev \
-  && rm -rf /var/lib/apt/lists/* \
-  && curl -L http://prdownloads.sourceforge.net/ta-lib/ta-lib-0.4.0-src.tar.gz \
-  | tar xvz \
-  && cd /ta-lib \
-  && ./configure --prefix=/usr \
-  && make \
-  && make install \
-  && pip install --upgrade pip
+        gfortran \
+        libfreetype6-dev \
+        libhdf5-dev \
+        liblapack-dev \
+        libopenblas-dev \
+        libpng-dev \
+    && rm -rf /var/lib/apt/lists/* \
+    && curl -fsSL http://prdownloads.sourceforge.net/ta-lib/ta-lib-0.4.0-src.tar.gz \
+    | tar xvz --strip-components 1 \
+    && ./configure --prefix="$TA_PREFIX" \
+    && make \
+    && make install
 
-WORKDIR /TA-Lib
+WORKDIR /src/ta-lib-python
 COPY . .
+RUN python -m pip install -e . \
+    && python -c 'import numpy, talib; close = numpy.random.random(100); output = talib.SMA(close); print(output)' \
```

Comment:
What is this line for?

Reply:
The `python -m pip install -e .` command installs the `ta-lib` python package. The next line came from the original Dockerfile and is basically a test that the package can be imported and can run `talib.SMA`.